src/main/kotlin/ecommerce/auth/JwtProvider.kt

```diff
@@ -0,0 +1,45 @@
+package ecommerce.auth
+
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import io.jsonwebtoken.security.Keys
+import java.util.*
+import javax.crypto.SecretKey
+
+object JwtProvider {
+    private val secretKey: SecretKey = Keys.secretKeyFor(SignatureAlgorithm.HS256)
+    private val EXPIRATION_TIME_MS = 3600_000 // 1 hour
+
+    fun generateToken(email: String): String {
+        val now = Date()
+        val expiry = Date(now.time + EXPIRATION_TIME_MS)
+
+        return Jwts.builder()
+            .setSubject(email)
+            .setIssuedAt(now)
+            .setExpiration(expiry)
+            .signWith(secretKey)
+            .compact()
+    }
+
+    fun validateToken(token: String): Boolean {
+        return try {
+            val claims = Jwts.parserBuilder()
+                .setSigningKey(secretKey)
+                .build()
+                .parseClaimsJws(token)
+            !claims.body.expiration.before(Date())
+        } catch (e: Exception) {
+            false
+        }
+    }
```

Comment:
```suggestion
    fun validateToken(token: String): Boolean {
        return try {
            val now = Date()
            val claims = Jwts.parserBuilder()
                .setSigningKey(secretKey)
                .build()
                .parseClaimsJws(token)
            claims.body.expiration.after(now)
        } catch (e: Exception) {
            false
        }
    }
```

`!` operator always make hard to read. 🤔